### PR TITLE
caps/eth.md: update to eth/65 and describe size limits

### LINKED ISSUE #154: ETH/65
Looks like we need to update the spec for the ETH protocol to reflect the changes made in https://eips.ethereum.org/EIPS/eip-2464

### PR DESCRIPTION
**Pull request by fjl**
Fixes #154

### CODE REVIEW

File: caps/eth.md [hunk 1]
```diff
@@ -157,6 +203,48 @@ specification.
...
+Ethereum specification.
+
+The transactions must be in same order as in the request, but it is OK to skip
+transactions which are not available. This way, if the response size limit is reached,
+requesters will know which hashes to request again (everything starting from the last
+returned transaction) and which to assume unavailable (all gaps before the last returned
+transaction).
+
+A peer may respond with an empty list iff none of the hashes match transactions in its
+pool. It is allowed to announce a transaction that will not be served later if it gets
```
Comment (fjl) [inline caps/eth.md:245]: @karalabe This sentence "...if it gets included in a block in between" is copied from the EIP, and the meaning isn't 100% clear to me. Can you explain what it means?

File: caps/eth.md [hunk 2]
```diff
@@ -1,49 +1,52 @@
...
+and clients typically perform them as concurrent activities on all peer connections.
 
-[Transactions] messages should also be sent periodically as the node has new transactions
-to disseminate. A node should never send a transaction back to a peer that it can
-determine already knows of it (either because it was previously sent or because it was
-informed from this peer originally).
+Client implementations should enforce limits on protocol message sizes. The underlying
+RLPx transport limits the size of a single message to 16.7 MiB. The practical limits for
+the eth protocol are lower, typically 10 MiB. If a received message is larger than the
+limit, the peer is disconnected.
```
Comment (holiman) [inline caps/eth.md:20]: perhaps "the peer is disconnected." -> "the peer should be disconnected."

File: caps/eth.md [hunk 3]
```diff
@@ -1,49 +1,52 @@
 # Ethereum Wire Protocol (ETH)
 
 'eth' is a protocol on the [RLPx] transport that facilitates exchange of Ethereum
-blockchain information between peers. The current protocol version is **eth/64**. See end
+blockchain information between peers. The current protocol version is **eth/65**. See end
```
Comment (holiman) [inline caps/eth.md:4]: Actually, although we haven't implemented in in geth, AFAIK `eth66` is the most recent approved protocol

File: caps/eth.md (hunk 1 above)
Comment (holiman) [reply caps/eth.md:245]: First peer A announces it, then it becomes included in a block. Peer A will clean it out from it's pool, and thus no longer have it accessible in memory. Since nodes are not expected to do disk-accesses to serve announce-responses, the node will not deliver it, despite earlier having announced it.

File: caps/eth.md (hunk 3 above)
Comment (holiman) [reply caps/eth.md:4]: See also https://github.com/ethereum/EIPs/issues/2482 and https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2481.md

Comment (fjl) [reply caps/eth.md:4]: Yes, that's correct, but the version here is the *spec version* and this PR updates it to eth/65.

File: caps/eth.md (hunk 1 above)
Comment (fjl) [reply caps/eth.md:245]: OK. Makes sense. I'll rephrase it to be more clear.

File: caps/eth.md [hunk 4]
```diff
@@ -52,6 +55,49 @@ a block near the head of the chain and downloads merkle tree nodes and contract
 incrementally by requesting the root node, its children, grandchildren, ... using
 [GetNodeData] until the entire tree is synchronized.
 
+### Block Propagation
+
+Newly-mined blocks must be relayed to all nodes. This happens through block announcements.
+
+Blocks are announced to all connected peers as soon as basic validity of the block has
```
Comment (karalabe) [inline caps/eth.md:62]: "announced to all connected"

This is false-ish. Blocks are propagated to sqrt(peers) after checking PoW. This ensures that the block ripples through the network if the PoW was valid. It however is not yet announced to everybody. After full block processing is finished and the block is saved to disk, only then is it announced. This is an important detail because the block **must** be in the database when announcing since we expect remote clients to immediately send requests for it that we need to be able to serve.

**Comment by karalabe** [comment]
LGTM
